### ISSUE TITLE
Log over sold items?

### ISSUE THREAD

**Issue by oband2** [kolton/d2bot-with-kolbot#1363]
Is it possible to have a log over sold items?
afaik there is just a log over kept and stashed items, but would like a log over sold items, so i can see if there is something in pickit i might want to change so i would keep those items instead.

**Comment by DarkHorseDre** [comment]
itemlog.txt logs all items processed including sold if you enable the right items in your *character config file*

`Config.ItemInfo = true;` // Log stashed, skipped (due to no space) or sold items.
`Config.ItemInfoQuality = [];`// The quality of sold items to log. See NTItemAlias.dbl for values. Example: Config.ItemInfoQuality = [6, 7, 8];

**Comment by oband2** [comment]
Ah thanks, didnt see the iteminfoquality. edited and added numbers, lets see what happens

**Comment by oband2** [comment]
Can i make it take screenshot of the item sold? like it takes screenshots in the manager and says whats kept and stashed?

**Comment by DarkHorseDre** [comment]
I dunno but look at the code where it takes screenshots.  

I searched the code and found references to screenshots, you can use the screenshot lines - the delay after may be for rendering - better to keep it:

if (location !== undefined) {
		D2Bot.printToConsole("Unhandled location " + location);
		takeScreenshot();
		delay(500);

**Comment by oband2** [comment]
> I dunno but look at the code where it takes screenshots.
> 
> I searched the code and found references to screenshots, you can use the screenshot lines - the delay after may be for rendering - better to keep it:
> 
> if (location !== undefined) {
> D2Bot.printToConsole("Unhandled location " + location);
> takeScreenshot();
> delay(500);

i guess you clicked on the wrong topic :D However if none knows the code, i think i rather skip it.

**Comment by oband2** [comment]
However this: 
Config.ItemInfo = true; // Log stashed, skipped (due to no space) or sold items.
It only logs skipped due to no space items, not sold :/

**Comment by 5noop** [comment]
You want to get a screenshot of every item the bot sells ? Because it's possible but there is gonna be a lot of items on your console.

**Comment by oband2** [comment]
@5noop;
Yes i would like to, just as i get screenshot of kept items. In The manager If possible so its Easy to view :-)

**Comment by 5noop** [comment]
Right, check my edit below then.

**Comment by oband2** [comment]
> Right, check my edit above then.

Looks like this now:
case 5: // Crafting System
						Misc.itemLogger("Kept", item, "CraftSys-Town");
						Misc.logItem("Sold", item);
						CraftingSystem.update(item);

Doesnt seem to work, should i do this instead?
case 5: // Crafting System
						Misc.itemLogger("Kept", "Sold", item, "CraftSys-Town");
						CraftingSystem.update(item);

**Comment by oband2** [comment]
none if it works :/ still only whats kept, just found a amulet and a dusk, both sold but no log
https://gyazo.com/86f60e75b7c84df0488172024b966640

**Comment by 5noop** [comment]
Yeah that's because you don't have the last version, or you already edited Town.js.
Check the last version to see which line I was talking about https://github.com/kolton/d2bot-with-kolbot/blob/master/d2bs/kolbot/libs/common/Town.js

**Comment by oband2** [comment]
I just copy pasted everything from your link, however it doesnt seem to work.
Do i need to uninstall whole bot and reinstall?
its only like 1 week ago since i installed the bot again

**Comment by 5noop** [comment]
No what I meant is you put it in the wrong line, maybe it's line 636 or something in your file.
Your code should look like that:
```
					case 5: // Crafting System
						Misc.itemLogger("Kept", item, "CraftSys-Town");
						CraftingSystem.update(item);

						break;
					default:
						Misc.logItem("Sold", item);
						Misc.itemLogger("Sold", item);
						item.sell();
```

**Comment by oband2** [comment]
Correct?:
https://gyazo.com/e6729013264f05eba899da56808fa24d

**Comment by 5noop** [comment]
Correct. Now save your file and type .reload in game, or just create a new game.

**Comment by oband2** [comment]
I did, saved it, restarted pc, started bot again, nothing :/

**Comment by 5noop** [comment]
No idea then, it works fine for me.

**Comment by oband2** [comment]
Can it have something to do with my char config? or is it only town.js that has anything to do with it?

**Comment by DarkHorseDre** [comment]
You should read the in-line notes for `Config.ItemInfoQuality` - it says to set a quality level as listed in `NTItemAlias.dbl` and I gave you 3 examples in my post above - your values are not correct, where did you get them from?
correct the values as per the list in that file:

NTIPAliasQuality["lowquality"] = 1;
NTIPAliasQuality["normal"] = 2;
NTIPAliasQuality["superior"] = 3;
NTIPAliasQuality["magic"] = 4;
NTIPAliasQuality["set"] = 5;
NTIPAliasQuality["rare"] = 6;
NTIPAliasQuality["unique"] = 7;
NTIPAliasQuality["crafted"] = 8;

In misc.js there is a section for logging items, and separate actions for sold, gambled etc.

var desc,
			date = new Date(),
			h = date.getHours(),
			m = date.getMinutes(),
			s = date.getSeconds(),
			dateString = "[" + (h < 10 ? "0" + h : h) + ":" + (m < 10 ? "0" + m : m) + ":" + (s < 10 ? "0" + s : s) + "]";

		switch (action) {
		case "Sold":
			if (Config.ItemInfoQuality.indexOf(unit.quality) === -1) {
				return false;
			}

			desc = this.getItemDesc(unit).split("\n").join(" | ").replace(/(\\xff|ÿ)c[0-9!"+<:;.*]/gi, "").trim();
//TAKE SCREENSHOT HERE
			break;
		case "Kept":

by default sold will be logged, so you can add the screenshot logic in there, but it may not be as straight forward.  get the simple logging in console working as we have told you here then look at the screenshots separately.  underatand the simple stuff and get them working first.

**Comment by oband2** [comment]
> You should read the in-line notes for `Config.ItemInfoQuality` - it says to set a quality level as listed in `NTItemAlias.dbl` and I gave you 3 examples in my post above - your values are not correct, where did you get them from?
> correct the values as per the list in that file:
> 
> NTIPAliasQuality["lowquality"] = 1;
> NTIPAliasQuality["normal"] = 2;
> NTIPAliasQuality["superior"] = 3;
> NTIPAliasQuality["magic"] = 4;
> NTIPAliasQuality["set"] = 5;
> NTIPAliasQuality["rare"] = 6;
> NTIPAliasQuality["unique"] = 7;
> NTIPAliasQuality["crafted"] = 8;
> 
> In misc.js there is a section for logging items, and separate actions for sold, gambled etc.
> 
> ```
> var desc,
> 			date = new Date(),
> 			h = date.getHours(),
> 			m = date.getMinutes(),
> 			s = date.getSeconds(),
> 			dateString = "[" + (h < 10 ? "0" + h : h) + ":" + (m < 10 ? "0" + m : m) + ":" + (s < 10 ? "0" + s : s) + "]";
> 
> 		switch (action) {
> 		case "Sold":
> 			if (Config.ItemInfoQuality.indexOf(unit.quality) === -1) {
> 				return false;
> 			}
> 
> 			desc = this.getItemDesc(unit).split("\n").join(" | ").replace(/(\\xff|ÿ)c[0-9!"+<:;.*]/gi, "").trim();
> //TAKE SCREENSHOT HERE
> 			break;
> 		case "Kept":
> ```
> by default sold will be logged, so you can add the screenshot logic in there, but it may not be as straight forward. get the simple logging in console working as we have told you here then look at the screenshots separately. underatand the simple stuff and get them working first.

Thanks for being more specific so that i understood correct, i obviously took wrong values, i took NTIPAliasType instead of quaility.

**Comment by oband2** [comment]
Logging works just fine now, just need the screenshot function now, iso codes for this :)

**Comment by 5noop** [comment]
Do you use CainID or FieldID ?

**Comment by oband2** [comment]
I use Vendors, mostly ormus in act3

**Comment by DarkHorseDre** [comment]
@5noop I see why you ask - needs to disable those id's else it will drop.  My town.js is adjusted to not drop but sell instead - did you do this?

**Comment by DarkHorseDre** [comment]
@5noop I see why you ask - needs to disable those id's else it will drop.  My town.js is adjusted to not drop but sell instead - did you do this?

@oband2 my 2nd post has the screenshot code example - did you read it?

**Comment by oband2** [comment]
> @5noop I see why you ask - needs to disable those id's else it will drop. My town.js is adjusted to not drop but sell instead - did you do this?
> 
> @oband2 my 2nd post has the screenshot code example - did you read it?

Yes this? but dont i need to edit something? im really bad at knowning what to change.
if (location !== undefined) {
D2Bot.printToConsole("Unhandled location " + location);
takeScreenshot();
delay(500);

**Comment by 5noop** [comment]
@DarkHorseDre actually it's very easy to produce an image of the dropped items too, but yeah the code I gave above should only work with sold items.

**Comment by oband2** [comment]
> @DarkHorseDre actually it's very easy to produce an image of the dropped items too, but yeah the code I gave above should only work with sold items.

hoooooow :(

**Comment by aim2kill** [comment]
@oband2 Look at this read legless las post
https://github.com/kolton/d2bot-with-kolbot/issues/250

**Comment by oband2** [comment]
> @oband2 read legless post, he show how to log drop and sold items
> #250
> With this you will be able to log and see the item screenshot in the console

Tried it, seems not, it seems it only logs sold uniques and sets, but it should log magic and rare aswell? or am i doing this wrong?
// Additional item info log settings. All info goes to \logs\ItemLog.txt
	Config.ItemInfo = true; // Log stashed, skipped (due to no space) or sold items.
	Config.ItemInfoQuality = [4, 5, 6, 7, 8]; // The quality of sold items to log. See NTItemAlias.dbl for values. Example: Config.ItemInfoQuality = [6, 7, 8];

**Comment by aim2kill** [comment]
Yes should log all you set in Config.ItemInfoQuality = [4, 5, 6, 7, 8]; 
I would recomment reinstalling or make a new copy and copy ur config,etc to the new copy and try by changing what have tell you in the previous post
